src/Microsoft.OpenApi/Validations/Rules/OpenApiSchemaRules.cs

```diff
@@ -70,15 +68,77 @@ public static class OpenApiSchemaRules
 
                     if (schema.Reference != null && schema.Discriminator != null)
                     {
-                        if (!schema.Required.Contains(schema.Discriminator?.PropertyName))
+                        var discriminator = schema.Discriminator?.PropertyName;
+                        var schemaReferenceId = schema.Reference.Id;
+
+                        if (!ValidateChildSchemaAgainstDiscriminator(schema, discriminator, schemaReferenceId, context))
                         {
                             context.CreateError(nameof(ValidateSchemaDiscriminator),
-                                                string.Format(SRResource.Validation_SchemaRequiredFieldListMustContainThePropertySpecifiedInTheDiscriminator,
-                                                                                schema.Reference.Id, schema.Discriminator.PropertyName));
+                            string.Format(SRResource.Validation_SchemaRequiredFieldListMustContainThePropertySpecifiedInTheDiscriminator,
+                                schemaReferenceId, discriminator));
                         }
                     }
 
                     context.Exit();
                 });
+
+        /// <summary>
+        /// Validates the property name in the discriminator against the ones present in the children schema
+        /// </summary>
+        /// <param name="schema">The parent schema.</param>
+        /// <param name="discriminator">Adds support for polymorphism. The discriminator is an object name that is used to differentiate
+        /// between other schemas which may satisfy the payload description.</param>
+        /// <param name="schemaReferenceId"></param>
+        /// <param name="context">A validation context.</param>
+        public static bool ValidateChildSchemaAgainstDiscriminator(OpenApiSchema schema, string discriminator, string schemaReferenceId, IValidationContext context)
+        {
+            bool containsDiscriminator = false;
+
+            if (!schema.Required.Contains(discriminator))
```

Comment:
```suggestion
            if (!schema.Required?.Contains(discriminatorName) ?? false)
```

Guard against potential null references.